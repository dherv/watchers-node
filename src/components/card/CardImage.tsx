import React from "react";
import styled from "styled-components";
import CardRating from "./CardRating";

const CardImage = ({ src, rating }: { src: string; rating: number }) => (
  <Container>
    <Image src={src}></Image>
    <CardRating size={48} rating={rating} />
  </Container>
);

const Container = styled.div`
  position: relative;
  width: 300px;
`;

const Image = styled.img``;
export default CardImage;
