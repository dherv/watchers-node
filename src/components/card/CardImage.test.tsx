import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import "jest-styled-components";

import CardImage from "./CardImage";

describe("CardImage", () => {
  const props = { src: "" };
  const wrapper = shallow(<CardImage {...props} />);

  describe("CardImage component", () => {
    describe("user interface", () => {
      test("should display one CardRating", () => {});
      test("should display an image with src = props.image_url", () => {});
    });
    describe("events", () => {});
  });

  describe("snapshot", () => {
    test("should match", () => {
      expect.assertions(1);
      const tree = renderer.create(<CardImage {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
