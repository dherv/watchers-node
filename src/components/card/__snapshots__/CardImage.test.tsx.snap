// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImage snapshot should match 1`] = `
.c1 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 5%;
  z-index: 10;
}

.c2 {
  position: relative;
  top: calc(20% - 67.88225099390857px / 2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 67.88225099390857px;
  height: 67.88225099390857px;
}

.c3 {
  position: relative;
  width: 48px;
  height: 48px;
  background-color: #2e8e89;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  z-index: 1;
}

.c4 {
  position: absolute;
  font-weight: 800;
  font-size: 1.2rem;
  color: #ececec;
  z-index: 10;
}

.c5 {
  position: absolute;
  width: 4px;
  background-color: var(--main-color-text);
  opacity: 0.8;
  z-index: 0;
  top: 0;
  left: 20%;
  height: 20%;
}

.c6 {
  position: absolute;
  width: 4px;
  background-color: var(--main-color-text);
  opacity: 0.8;
  z-index: 0;
  bottom: 0;
  right: 25%;
  height: 80%;
}

.c0 {
  position: relative;
  width: 300px;
}

<div
  className="c0"
>
  <img
    className=""
    src=""
  />
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        size={48}
      />
      <h5
        className="c4"
      />
    </div>
    <div
      className="c5"
    />
    <div
      className="c6"
    />
  </div>
</div>
`;
